src/server/chromium/crPage.ts

```diff
@@ -93,7 +93,13 @@ export class CRPage implements PageDelegate {
     }
     // Note: it is important to call |reportAsNew| before resolving pageOrError promise,
     // so that anyone who awaits pageOrError got a ready and reported page.
-    this._pagePromise = this._mainFrameSession._initialize(hasUIWindow).then(() => {
+    this._pagePromise = this._mainFrameSession._initialize(hasUIWindow).then(async r => {
+      await this._page.initOpener(this._opener);
```

Comment:
This should be in page.ts because it is the same across ports?

Reply:
I considered placing it into `Page.reportAsNew` but it would require making that method async which I'd like to avoid and keep it sending sync events only. This code needs more cleanup anyway, I can fix that separately.